src/Serilog.Sinks.AzureDataExplorer/Sinks/AzureDataExplorerDurableSink.cs

```diff
@@ -0,0 +1,187 @@
+using System.Text;
+using Kusto.Data.Common;
+using Kusto.Ingest;
+using Serilog.Core;
+using Serilog.Events;
+using Serilog.Formatting.Compact;
+using Serilog.Sinks.AzureDataExplorer.Durable;
+using Serilog.Sinks.AzureDataExplorer.Extensions;
+
+namespace Serilog.Sinks.AzureDataExplorer.Sinks
+{
+    internal class AzureDataExplorerDurableSink : ILogEventSink, IDisposable
+    {
+        const string FileNameSuffix = "-.clef";
+        private static readonly List<ColumnMapping> SDefaultIngestionColumnMapping = new List<ColumnMapping>
+        {
+            new ColumnMapping
+            {
+                ColumnName = "Timestamp",
+                ColumnType = "datetime",
+                Properties = new Dictionary<string, string>
+                {
+                    {
+                        MappingConsts.Path, "$.Timestamp"
+                    }
+                }
+            },
+            new ColumnMapping
+            {
+                ColumnName = "Level",
+                ColumnType = "string",
+                Properties = new Dictionary<string, string>
+                {
+                    {
+                        MappingConsts.Path, "$.Level"
+                    }
+                }
+            },
+            new ColumnMapping
+            {
+                ColumnName = "Message",
+                ColumnType = "string",
+                Properties = new Dictionary<string, string>
+                {
+                    {
+                        MappingConsts.Path, "$.Message"
+                    }
+                }
+            },
+            new ColumnMapping
+            {
+                ColumnName = "Exception",
+                ColumnType = "string",
+                Properties = new Dictionary<string, string>
+                {
+                    {
+                        MappingConsts.Path, "$.Exception"
+                    }
+                }
+            },
+            new ColumnMapping
+            {
+                ColumnName = "Properties",
+                ColumnType = "dynamic",
+                Properties = new Dictionary<string, string>
+                {
+                    {
+                        MappingConsts.Path, "$.Properties"
+                    }
+                }
+            },
+        };
+
+        private readonly Logger m_sink;
+
+        private readonly IKustoQueuedIngestClient m_ingestClient;
+        private readonly LogShipper<List<LogEvent>> m_shipper;
+
+        public AzureDataExplorerDurableSink(AzureDataExplorerSinkOptions options)
+        {
+            if (options == null)
```

Comment:
this.options= options ?? throw new ArgumentNullException(nameof(options)) is a shorthand for this

Reply:
used shorthand wherever possible. seems like in C# , for 'this' to be used, options need to be defined in the scope of the class where this is used. It is not picking the method scope.